include/af/algorithm.h

```diff
@@ -85,6 +150,24 @@ namespace af
     */
     AFAPI array min(const array &in, const int dim = -1);
 
+#if AF_API_VERSION >= 37
+    /**
+       C++ Interface for minimum values in an array according to a key
+
+       \param[out] keys_out will contain the reduced keys in \p vals along \p dim
+       \param[out] vals_out will contain the minimum of all values in \p vals along \p dim according to \p keys
+       \param[in] keys is the key array
+       \param[in] vals is the array containing the values to be reduced
+       \param[in] dim The dimension along which the add operation occurs
+
+       \ingroup reduce_func_min
+
+       \note \p dim is -1 by default. -1 denotes the first non-singleton dimension.
+       \note NaN values are ignored
+    */
+    AFAPI void minByKey(array &keys_out, array &vals_out, const array &keys, const array &vals, const int dim = -1);
+#endif
+
```

Comment:
This too, Lines 153-138 can we also break this up into 80 chars per line? Or I suppose running `clang-format` here would help here and in the previous chunks that needs to be broken as well.